Gutenlypso: Update edit-post to the latest version

Copons opened the pull request:
#### Changes proposed in this Pull Request

* Update the `edit-post` folder with the latest and greatest [Core `edit-post` package](https://github.com/WordPress/gutenberg/tree/master/packages/edit-post/src/).

All the changes and additions to the `edit-post` folder are marked with a `GUTENLYPSO` comment.
The deletions are unfortunately not marked (my bad, I noticed it way too late).

#### Freebies

The block navigation:
![screenshot 2018-12-04 at 17 58 15](https://user-images.githubusercontent.com/2070010/49462740-b07c1880-f7ee-11e8-930f-aad9837cb57f.png)

The more menu (now 99% identical to Core, and moved back into the `edit-post` folder):
![screenshot 2018-12-04 at 17 58 31](https://user-images.githubusercontent.com/2070010/49462762-bf62cb00-f7ee-11e8-8497-e0afce32ae88.png)

The options modal (without the "enable NUX tips" options):
![screenshot 2018-12-04 at 17 58 36](https://user-images.githubusercontent.com/2070010/49462802-e02b2080-f7ee-11e8-83d1-49beb1a66ca5.png)

The keyboard shortcuts modal (all of them _should_ be working; some that weren't working before are working now though):
![screenshot 2018-12-04 at 17 58 50](https://user-images.githubusercontent.com/2070010/49462890-1e284480-f7ef-11e8-86fd-c4a3927ba80f.png)

The preview now has the nice "generating" animation 💖:
![screenshot 2018-12-04 at 17 59 51](https://user-images.githubusercontent.com/2070010/49462912-284a4300-f7ef-11e8-8ee8-c921815a1b1e.png)

#### Testing instructions

Make sure there are no obvious regressions on `/gutenberg`.

Pay special attention to:
- How the header looks.
- The preview and publish flows.
- The more menu (e.g. the Keyboard Shortcuts and Options modals).
- The sidebar.


Fixes #28989 #28990

kwight commented:
ZOMG this is _soooooooooo_ nice. Noticed a few things while comparing this branch to a fresh install of 5.0 beta + JP with jurassic.ninja:

**Block navigation:**
- Calypso has extra list bullets:
<img width="399" alt="screen shot 2018-12-04 at 11 35 46 am" src="https://user-images.githubusercontent.com/349751/49468169-02a24700-f7b9-11e8-9fda-a0e8311a7492.png">


**More menu:** 
- Calypso doesn't have "Fullscreen" (which I think is because we only have fullscreen editing in Calypso?).
- Manage Reusable Blocks is missing from Calypso (also good I think?)

**Contact Form:**
- Calypso doesn't have red styling of "(required)" text:
<img width="914" alt="screen shot 2018-12-04 at 11 35 18 am" src="https://user-images.githubusercontent.com/349751/49468139-eef6e080-f7b8-11e8-9365-761201dfd46d.png">

dmsnell commented:
I believe this is unrelated, but I noticed that the revisions dialog is escaping where it shouldn't be; you can see the escaped double-quotes in the JSON attributes that should be actual quotes and not `&quot;`

<img width="615" alt="screen shot 2018-12-04 at 2 06 19 pm" src="https://user-images.githubusercontent.com/5431237/49473051-3b98e680-f7ce-11e8-9516-8a3d0b2f661f.png">

(Confirmed - this is on `master` outside of Gutenberg as well)

kwight commented:
Note this'll need a rebase if #29094 gets in first (but that PR should solve some of what I saw with the contact form).

Copons commented:
- Calypso has extra list bullets

Fixed! I didn't know the block navigation could have nested lists.

- Calypso doesn't have "Fullscreen" (which I think is because we only have fullscreen editing in Calypso?).
- Manage Reusable Blocks is missing from Calypso (also good I think?)

As intended. Guten is only fullscreen in Calypso, and doesn't have a manage reusable blocks section (yet?).

- Calypso doesn't have red styling of "(required)" text:

Fixed by #29094

- revisions dialog is escaping where it shouldn't be

Unrelated, we literally import the Classic editor revision dialog 🙂
